Survey/Survey/Sources/Data/Models/LoginTokenApi.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  LoginTokenApi.swift
+//  Survey
+//
+//  Created by Taher on 4/11/22.
+//
+
+import Foundation
+
+struct LoginTokenApi: Decodable, LoginToken {
+
+    enum CodingKeys: String, CodingKey {
+        case data
+        case attributes
+        case accessToken = "access_token"
+        case tokenType = "token_type"
+        case expiresIn = "expires_in"
+        case refreshToken = "refresh_token"
+        case createdAt = "created_at"
+    }
+
+    let accessToken: String
+    let tokenType: String
+    let expiresIn: Int
+    let refreshToken: String
+    let createdAt: Int
+
+    init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        let data = try container.nestedContainer(keyedBy: CodingKeys.self, forKey: .data)
+        let attributes = try data.nestedContainer(keyedBy: CodingKeys.self, forKey: .attributes)
+        accessToken = try attributes.decode(String.self, forKey: .accessToken)
+        tokenType = try attributes.decode(String.self, forKey: .tokenType)
+        expiresIn = try attributes.decode(Int.self, forKey: .expiresIn)
+        refreshToken = try attributes.decode(String.self, forKey: .refreshToken)
+        createdAt = try attributes.decode(Int.self, forKey: .createdAt)
+    }
+}
```

Comment:
Since our API's response conform to [JSONAPI](https://jsonapi.org) spec, we can leverage it by using a library for parsing instead of doing so manually. A good one is our Nimble's [JSONAPIMapper](https://github.com/nimblehq/JSONMapper)

Reply:
nice suggestion, I didn't aware of that. I created a new [issue](https://github.com/Shayokh144/ic-survey-swiftui-combine/issues/9) for next sprint as there are some remaining task in current sprint.